### PR TITLE
Polish outdated configs

### PR DESCRIPTION
**Pull request by Goooler**
Follow up 1a57c16d8eacfc412e57370011272319844496bf.

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -52,17 +54,3 @@ sourceSets {
...
-        options.compilerArgs << '-proc:none'
-    }
-}
-
-tasks.withType(GroovyCompile).configureEach {
-    // This will be the default in Gradle 5.0
-    if (!options.compilerArgs.contains("-processor")) {
-        options.compilerArgs << '-proc:none'
-    }
-}
```
Comment (Goooler) [inline buildSrc/build.gradle:68]: Seems we can remove these from `buildSrc` too?

File: build.gradle [hunk 2]
```diff
@@ -1,21 +1,16 @@
...
-    // id 'signing'
     id 'com.gradle.plugin-publish' version '1.1.0'
     id 'org.ajoberstar.git-publish' version '4.1.1'
     id 'com.github.node-gradle.node' version '3.5.1'
 }
 
-// Remove the gradleApi so it isn't merged into the jar file.
-configurations.named(JavaPlugin.API_CONFIGURATION_NAME) {
-    dependencies.remove(project.dependencies.gradleApi())
-}
```
Comment (Goooler) [inline build.gradle:17]: `compileOnly` should be enough

```sh
./gradlew assemble

diffuse diff --jar shadow-8.1.0-SNAPSHOT-before.jar shadow-8.1.0-SNAPSHOT-after.jar

OLD: shadow-8.1.0-SNAPSHOT-before.jar
NEW: shadow-8.1.0-SNAPSHOT-after.jar

 JAR   │ old       │ new       │ diff 
───────┼───────────┼───────────┼──────
 class │ 700.9 KiB │ 700.9 KiB │  0 B 
 other │  10.6 KiB │  10.6 KiB │  0 B 
───────┼───────────┼───────────┼──────
 total │ 711.5 KiB │ 711.5 KiB │  0 B 


 CLASSES │ old  │ new  │ diff      
─────────┼──────┼──────┼───────────
 classes │  146 │  146 │ 0 (+0 -0) 
 methods │ 1813 │ 1813 │ 0 (+0 -0) 
  fields │  746 │  746 │ 0 (+0 -0) 
```

File: gradle/docs.gradle [hunk 3]
```diff
@@ -9,13 +9,13 @@ tasks.withType(Javadoc).configureEach {
...
-tasks.register('javadocJar', Jar) {
-    dependsOn javadoc
+def javadocJar = tasks.register('javadocJar', Jar) {
+    dependsOn tasks.withType(Javadoc)
     archiveClassifier.set('javadoc')
     from 'build/docs/javadoc'
 }
 
-tasks.register('sourcesJar', Jar) {
+def sourcesJar = tasks.register('sourcesJar', Jar) {
```
Comment (johnrengelman) [inline gradle/docs.gradle:18]: I think we can drop these tasks entirely and instead use
```
java {
    withJavadocJar()
    withSourcesJar()
}
```

File: build.gradle [hunk 4]
```diff
@@ -1,21 +1,16 @@
 import com.github.jengelman.gradle.plugins.shadow.ShadowPlugin
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
 
 plugins {
     id 'groovy'
     id 'project-report'
     id 'idea'
     id 'java-gradle-plugin'
-    // id 'signing'
```
Comment (johnrengelman) [inline build.gradle:8]: I think I will re-add `signing` as `plugin-publish` supports it natively. I had turned this off when troubleshooting errors with publishing the 8.1.0 plugin.

File: gradle/docs.gradle (hunk 3 above)
Comment (Goooler) [reply gradle/docs.gradle:18]: Good catch!

File: build.gradle (hunk 4 above)
Comment (Goooler) [reply build.gradle:8]: I'll put it back, believe you just set `signing.required = false` will disable the sign tasks.

File: gradle/publish.gradle [hunk 5]
```diff
@@ -41,21 +35,22 @@ tasks.named('publishPlugins') {
...
-// }
+signing {
+    if (System.env.CI == 'true') {
+        def encodedSigningKey = findProperty("signingKey")
+        def signingKey = encodedSigningKey ? new String(encodedSigningKey.decodeBase64()) : null
+        def signingPassword = findProperty("signingPassword") as String
+        useInMemoryPgpKeys(signingKey, signingPassword)
+    }
+    // See https://github.com/johnrengelman/shadow/pull/831#discussion_r1119012328
+    required = false && gradle.taskGraph.hasTask("artifactoryPublish")
```
Comment (Goooler) [inline gradle/publish.gradle:46]: Disabled for now.

File: build.gradle (hunk 4 above)
Comment (Goooler) [reply build.gradle:8]: https://github.com/johnrengelman/shadow/pull/831#discussion_r1119028819

**Review by johnrengelman — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (johnrengelman) [reply build.gradle:17]: I've read-ed this line in 8.1.1. I'm not sure if that affects the configuration caching in an way.

**Pull request merged**